src/adaXT/predict/predict.pyx

```diff
@@ -3,7 +3,64 @@ import numpy as np
 from numpy import float64 as DOUBLE
 from ..decision_tree.nodes import DecisionNode
 from collections.abc import Sequence
+from statistics import mode
 cimport numpy as cnp
+from adaXT.parallel import shared_numpy_array
+
+
+def default_predict(tree, X, **kwargs):
+    return np.array(tree.predict(X, **kwargs))
+
+
+def predict_proba(tree, Y, X, unique_classes):
```

Comment:
nit but maybe call proba_predict so its inline with the other predict functions outside the predict class

Reply:
not sure I follow.

Reply:
I just just meant to be consistent with having the "predict" part of the naming last (as this is how its currently done for the other functions, e.g., quantile_predict).